### PR TITLE
Make subscript and superscript text formats mutually exclusive

### PR DESCRIPTION
**Pull request by geryogam**
Closes https://github.com/facebook/lexical/issues/5303

### CODE REVIEW

File: packages/lexical/src/nodes/LexicalTextNode.ts [hunk 1]
```diff
@@ -702,7 +703,13 @@ export class TextNode extends LexicalNode {
    */
   toggleFormat(type: TextFormatType): this {
     const formatFlag = TEXT_TYPE_TO_FORMAT[type];
-    return this.setFormat(this.getFormat() ^ formatFlag);
+    let newFormat = this.getFormat() ^ formatFlag;
```
Comment (zurfyx) [inline packages/lexical/src/nodes/LexicalTextNode.ts:706]: Can we have this function calling `toggleTextFormatType` since they're roughly the same?

**Review by zurfyx — APPROVED** [review]
LGTM, thank you! Do you mind adding a unit test?

**Comment by geryogam** [comment]
@zurfyx I made the requested changes, could you review again?

**Comment by zurfyx** [comment]
Thanks for the updates!

**Comment by geryogam** [comment]
Thanks for the review!

**Pull request merged**